transport_udp.go

```diff
@@ -73,42 +75,45 @@ func NewUDPTransport(hostPort string, maxPacketSize int) (transport.Transport, e
 
 	sender := &udpSender{
 		client:         client,
-		maxSpanBytes:   maxPacketSize - emitSpanBatchOverhead,
+		maxSpanBytes:   maxPacketSize - emitBatchOverhead,
 		thriftBuffer:   thriftBuffer,
 		thriftProtocol: thriftProtocol}
 	return sender, nil
 }
 
-func (s *udpSender) calcSpanSize(span *zipkincore.Span) (int, error) {
+func (s *udpSender) calcSizeOfSerializedThrift(thriftStruct thrift.TStruct) int {
 	s.thriftBuffer.Reset()
-	if err := span.Write(s.thriftProtocol); err != nil {
-		return 0, err
-	}
-	return s.thriftBuffer.Len(), nil
+	thriftStruct.Write(s.thriftProtocol)
+	return s.thriftBuffer.Len()
 }
 
-func (s *udpSender) Append(span *zipkincore.Span) (int, error) {
-	spanSize, err := s.calcSpanSize(span)
-	if err != nil {
-		// should not be getting this error from in-memory transport - ¯\_(ツ)_/¯
-		return 1, err
+func (s *udpSender) Append(span *Span) (int, error) {
+	if s.process == nil {
+		s.process = &j.Process{
+			ServiceName: span.tracer.serviceName,
+			Tags:        buildTags(span.tracer.tags),
+		}
+		s.processByteSize = s.calcSizeOfSerializedThrift(s.process)
+		s.byteBufferSize += s.processByteSize
```

Comment:
fortunately, the struct overhead + process tags overhead is still < 30 (it's 26 according to the tests)